packages/bitcore-node/src/modules/ripple/api/csp.ts

```diff
@@ -91,16 +92,41 @@ export class RippleStateProvider extends InternalStateProvider implements IChain
     }
   }
 
+  async getWalletBalanceAtTime(params: GetWalletBalanceAtTimeParams) {
+    const { chain, network, time } = params;
+    const addresses = await this.getWalletAddresses(params.wallet._id!);
+    const balances = await Promise.all(
+      addresses.map(a => this.getBalanceForAddress({ address: a.address, chain, network, args: { time } }))
+    );
+    return balances.reduce(
+      (total, current) => {
+        total.balance += current.balance;
+        total.confirmed += current.confirmed;
+        total.unconfirmed += current.unconfirmed;
+        return total;
+      },
+      { confirmed: 0, unconfirmed: 0, balance: 0 }
+    );
+  }
+
   async getBalanceForAddress(params: GetBalanceForAddressParams) {
-    const { chain, network, address } = params;
+    const { chain, network, address, args } = params;
     const lowerAddress = address.toLowerCase();
     const cacheKey = `getBalanceForAddress-${chain}-${network}-${lowerAddress}`;
     return CacheStorage.getGlobalOrRefresh(
       cacheKey,
       async () => {
         const client = await this.getClient(network);
         try {
-          const balance = await client.getBalance({ address });
+          let ledgerIndex: number | undefined;
+          if (args?.time) {
```

Comment:
This gets cached under the same non-time key. Need to check if this is also true for the other chains affected in this PR (EVM, SVM)